### ISSUE TITLE
v0.7.1 tag doesn't deploy `ReferenceGrant` CRD via Kustomize

### ISSUE THREAD

**Issue by hawksight** [kubernetes-sigs/gateway-api#2339]
**What happened**:

When you download the v0.7.1 standard release yaml and extract just the CRD resource, there are 4 of them:

```shell
yq ea 'select(.kind == "CustomResourceDefinition") | .metadata.name' ~/Downloads/gateway-api-standard-install-v0.7.1.yaml 
gatewayclasses.gateway.networking.k8s.io
---
gateways.gateway.networking.k8s.io
---
httproutes.gateway.networking.k8s.io
---
referencegrants.gateway.networking.k8s.io
```

However when you run the following command it only returns 3 resources:

```shell
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.7.1" | yq ea '.metadata.name' -
gatewayclasses.gateway.networking.k8s.io
---
gateways.gateway.networking.k8s.io
---
httproutes.gateway.networking.k8s.io
```

The issue is that the kustomization file [here](https://github.com/kubernetes-sigs/gateway-api/blob/v0.7.1/config/crd/kustomization.yaml) does not include that resource, only the 3 other CRDs.

Note that [release-0.7 branch](https://github.com/kubernetes-sigs/gateway-api/blob/release-0.7/config/crd/kustomization.yaml) is up to date with the relevant line added.

The issue is that some other provided reference the git branch directly in their documentation with `kustomize` such as in Openshift ServiceMesh [here](https://access.redhat.com/documentation/en-us/red_hat_openshift_service_on_aws/4/html/service_mesh/service-mesh-2-x#installing-the-gateway-api-crds). Note that yes they suggest v0.4.0 however I wanted to apply the latest.

**What you expected to happen**:

I expected both the release and then tag would have the same behaviour. I think [this commit](https://github.com/kubernetes-sigs/gateway-api/commit/aa1984fbe4b669b02d20c10b5e60a88feeb527f8) just needs ported to the v0.7.1 tag? or a new v0.7.2 release might be needed?

**How to reproduce it (as minimally and precisely as possible)**:

Not working

```shell
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.7.1" | yq ea '.metadata.name' -
```

Works:

```shell
curl https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.7.1/standard-install.yaml -L -O
yq ea 'select(.kind == "CustomResourceDefinition") | .metadata.name' ./standard-install.yaml
```

**Anything else we need to know?**:

This could be fixed in the RedHat docs for Openshift, but it still feels inconsistent here.

Also, this also fixes it by using the release branch as the ref instead:

```shell
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=release-0.7" | yq ea '.metadata.name' -
```

**Comment by hawksight** [comment]
Also for reference, if people were following [istio docs](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#setup) but tweaking version, they might also encounter this issue.

**Comment by robscott** [comment]
Thanks for reporting this @hawksight!

This was fixed by https://github.com/kubernetes-sigs/gateway-api/pull/2084 and will be released in v0.8.0 (including the existing RC1 release).
